File: app/src/main/java/it/amonshore/comikkua/ui/IViewHolderWithDetails.java
package it.amonshore.comikkua.ui;

import android.view.View;

import androidx.recyclerview.selection.ItemDetailsLookup;
import androidx.recyclerview.widget.RecyclerView;

public abstract class IViewHolderWithDetails<K> extends RecyclerView.ViewHolder {

    public IViewHolderWithDetails(View itemView) {
        super(itemView);
    }

    public abstract ItemDetailsLookup.ItemDetails<K> getItemDetails();

}

File: app/src/main/java/it/amonshore/comikkua/ui/IViewHolderWithDetails.kt
package it.amonshore.comikkua.ui

import android.view.View
import androidx.recyclerview.selection.ItemDetailsLookup.ItemDetails
import androidx.recyclerview.widget.RecyclerView

abstract class IViewHolderWithDetails<K>(itemView: View) : RecyclerView.ViewHolder(itemView) {
    abstract val itemDetails: ItemDetails<K>?
}